src/main/java/pl/kinga/produkty/ProductController.java

```diff
@@ -0,0 +1,55 @@
+package pl.kinga.produkty;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+public class ProductController {
+    ProductRepository productRepository;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/")
+    public String home(Model model, @RequestParam(required = false) ProductCategory category) {
+        return "home";
+    }
+
+    @GetMapping("/allProducts")
+    public String product(@RequestParam(required = false, value = "category") ProductCategory category, Model model) {
+        List<Product> products = productRepository.findAll();
+
+        if (category != null) {
+            model.addAttribute("product", new Product());
+            model.addAttribute("products", productRepository.findByCategory(category));
+        } else {
+            model.addAttribute("products", productRepository.findAll());
+        }
```

Comment:
Cały kod metody powyżej możesz przerobić do takiej formy:
```
        List<Product> products;
        
        if (category != null) {
            products = productRepository.findByCategory(category);
        } else {
            products = productRepository.findAll();
        }

        model.addAttribute("products", products);
```
Co to nam daje?
1. Wywolywana jest tylko jedna mtoda repozytorium dla jednego wywołania metody `product`. Poki co repozytorium odwoluje sie do listy znajdującej się w pamięci, ale w praktyce (i w późniejszych zadaniach) metody repozytoriów budują i wykonują zapytania do baz danych, które są kosztowne.
2. W pętli niżej do summowania cen nie będziesz potrzebować w ogóle ifa